### PR TITLE
Allow a has_many association to specify an explicit foreign key via the :as option

### PR DESCRIPTION
**Pull request by mbklein**
Fixes the bug demonstrated in [this gist](https://gist.github.com/mbklein/88d07b3fe6ad97ee5f1d), which I never got around to filing an issue about because I fixed it instead.

### CODE REVIEW

File: spec/integration/associations_spec.rb [hunk 1]
```diff
@@ -24,6 +24,37 @@ class Person < ActiveFedora::Base
     end
 
   end
+  
+  describe "explicit foreign key" do
+    before do
+      class FooThing < ActiveFedora::Base
+        has_many :bars, :class_name=>'BarThing', :predicate=>ActiveFedora::RDF::Fcrepo::RelsExt.isPartOf, :as=>:foothing
```
Comment (awead) [inline spec/integration/associations_spec.rb]: Do we want to make "foothing" implicit, but overrideable? If the class was named "Page" then it seems implicit that `:as => :page`.

Comment (mbklein) [inline spec/integration/associations_spec.rb]: This is testing the override. The implicit key was `foo_thing`

Comment (awead) [inline spec/integration/associations_spec.rb]: :ok:

Comment (mbklein) [inline spec/integration/associations_spec.rb]: :cool:

**Pull request merged**